File: 62.不同路径.js
/*
 * @lc app=leetcode.cn id=62 lang=javascript
 *
 * [62] 不同路径
 */

// @lc code=start
/**
 * @param {number} m
 * @param {number} n
 * @return {number}
 */
var uniquePaths = function (m, n) {
    return factorial(m + n - 2) / (factorial(m - 1) * factorial(n - 1))
};

function factorial(n) {
    if (n === 0 || n === 1) {
        return 1;
    } else {
        return n * factorial(n - 1);
    }
}
// @lc code=end

console.log(uniquePaths(3, 2));
console.log(uniquePaths(7, 3));

File: 62.不同路径.py
#
# @lc app=leetcode.cn id=62 lang=python3
#
# [62] 不同路径
#

# @lc code=start
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        f = [[0] * n] * m
        for i in range(0, m):
            for j in range(0, n):
                if i == 0 or j == 0:
                    f[i][j] = 1
                else:
                    f[i][j] = f[i-1][j] + f[i][j-1]
        return f[m-1][n-1]

# @lc code=end


solution = Solution()
res = solution.uniquePaths(3, 2)
print(res)
